GitHubRepositories.ps1

```diff
@@ -106,6 +115,9 @@ function New-GitHubRepository
 
         [switch] $Private,
 
+        [ValidateSet('Public', 'Private', 'Internal')]
+        [string] $Visibility,
```

Comment:
In an ideal world, we'd only have this and wouldn't have `[switch] $Private`, because this can lead to confusion, but dropping `[switch] $Private` would be a breaking change.  So, let's just do some safe-guarding.

1. Please add a check in the body that's something similar to:
```powershell
    if ($PSBoundParameters.ContainsKey('Private'))
    {
        Write-Log -Level Warning -Message 'The Private switch is being replaced by the Visibility parameter.  Please update any scripts/usage that you may have to migrate to the Visibility parameter, as the Private switch will be replaced in a future update.'

        if ($PSBoundParameters.ContainsKey('Visibility'))
        {
            if (($Private -and ($Visibility -ne 'Private')) -or ((-not $Private) -and ($Visibility -ne 'Public')))
            {
                $message = 'The Private and Visibility parameters control the same repo property.  If both are specified, their values must be the same.'
                Write-Log -Message $message -Level Error
                throw $message
            }
        }
    }
```

2. Please enhance the `Visibility` CBH description to explain that `Internal` is only available `if your organization is associated with an enterprise account using GitHub Enterprise Cloud or GitHub Enterprise Server 2.20+`, and that it controls the same property as the `Private` switch.

3. Please update the `Private` switch CBH description to explain that it controls the same property as `Visibility` and that only one should be used.

Reply:
Similar feedback for the `Update-GitHubRepository`

Reply:
Done

Reply:
Looks like you missed the one in GitHubProjects.tests.ps1:

https://github.com/microsoft/PowerShellForGitHub/blob/00f099fa81d25908735dc32d86736180f6209d34/Tests/GitHubProjects.tests.ps1#L204

Reply:
That's a different function though...

Reply:
<sigh> Completely missed that.  Apologies.